src/Projects/IrcClient/Clients/DccClient.cs

```diff
@@ -25,7 +25,8 @@ public DccClient(string filename, ulong filesize, int buffersize = 4096)
 
         public event Action<ulong, ulong> Progress;
 
-        public async Task Connect(string host, int port)
+        // TODO: use CancellationToken for socket when it's implemented in .net
+        public async Task Connect(string host, int port, CancellationToken token)
```

Comment:
Dotnet 5 has an overload with CancellationToken I think, so when/if we move to dotnet 5, we can change this